### PR TITLE
improvements from Moonlight

### PR DESCRIPTION
**Pull request by ehartford**
MoonshotAI suggested some improvements to Muon in training Moonlight.

https://github.com/MoonshotAI/Moonlight/blob/master/examples/toy_train.py

Perhaps they would make a good addition to Muon?

### CODE REVIEW

**Comment by toothacher17** [comment]
hi, @ehartford thanks for mentioning it! To be fair, I think the core ideas of training Moonlight are:

1. weight decay
2. adjusting update rms by matrix shape
3. matching to AdamW RMS

For point 1 and 2, Keller's current impl should already contain it (as we mentioned in the paper). For point 3, it might not be the best setting under the nanogpt speedrun

**Comment by lin72h** [comment]
@toothacher17 Nice suggestion!

**Comment by KellerJordan** [comment]
Hello -- It is my belief that the current Muon implementation already uses an ideal learning rate scaling formula (`max(1, out_dim/in_dim)**0.5`), in the sense that this formula should imply a constant optimal learning rate as a function of scale. As @toothacher17 mentioned, the other changes are already incorporated.
